labcontrol/db/process.py

```diff
@@ -3819,3 +3826,407 @@ def _generate_amplicon_prep_information(self):
+            item['well_id'] = ''.join(well) + str(col)
+
+            # Note: currently we have reverted to generating just one prep
+            # sheet for all items in run, but we anticipate that may change
+            # back in the near future.  That is why we retain the return
+            # structure of a dictionary holding prep sheet strings rather
+            # than returning a single prep sheet string even though, at the
+            # moment, the dictionary will always have only one entry.
+            curr_prep_sheet_id = self.run_name
+            if curr_prep_sheet_id not in data:
+                data[curr_prep_sheet_id] = {}
+
+            content = item['content']
+
+            # adding item to the data (organized by prep_sheet_id and
+            # content string.
+            if content in data[curr_prep_sheet_id]:
+                raise ValueError("'%s' appears more than once" % content)
+
+            data[curr_prep_sheet_id][content] = item
+
+        # right now, there will only be one prep_sheet_id
+        for prep_sheet_id, prep_sheet in data.items():
+            prep_sheet = pd.DataFrame.from_dict(prep_sheet, orient='index')
+
+            # If orig_name2 is none (because this item is a control),
+            # use its content. Note that if v (the value of orig_name2 at that
+            # row) is None, then the value for orig_name2 at that row will
+            # become the value of the index (k) for that row. Note that we
+            # are not currently using the value of 'is_control'.
+            prep_sheet.orig_name2 = [v if v else k for k, v in
+                                     prep_sheet.orig_name2.iteritems()]
+
+            # Set the project column value for each non-experimental sample to
+            # the value of the project name for the (single) qiita study on
+            # that sample's plate.
+            prep_sheet =\
+                self._set_control_values_to_plate_value(prep_sheet,
+                                                        'sample_plate',
+                                                        'project_name')
+
+            # mapping keys to expected names for columns in the final output
+            mv = {""
+                  "orig_name2": "Orig_name",
+                  "well_id": "Well_ID",
+                  "sample_plate": "Sample_Plate",
+                  "project_name": "Project_name",
+                  "plating": "Plating",
+                  "barcode_i7": "index",
+                  "barcode_i5": "index2",
+                  "primer_plate_i7": "i7_Primer_Plate",
+                  "primer_plate_i5": "i5_Primer_Plate",
+                  "primer_date_i7": "i7_Primer_date",
+                  "primer_date_i5": "i5_Primer_date",
+                  "experiment_design_description":
+                      "EXPERIMENT_DESIGN_DESCRIPTION",
+                  "instrument_model": "INSTRUMENT_MODEL",
+                  "kapa_hyper_plus_kit_lot": "KapaHyperPlusKit_lot",
```

Comment:
Would it be possible to add an inline comment noting #503? It would be wonderful for #503 to have a permalink to this line as well, may reduce some confusion later when resolving that item.